packages/snjs/lib/services/api/http_service.ts

```diff
@@ -89,6 +90,21 @@ export class SNHttpService extends PureService {
     }
     request.open(httpRequest.verb, httpRequest.url, true);
     request.setRequestHeader('Content-type', 'application/json');
+
+    // TODO: `localStorage` might not be present in mobile, perhaps need to use the service instead
+    const environmentWithVersion = localStorage.getItem('environmentWithVersion');
```

Comment:
Never access `localStorage` directly from SNJS. SNJS is _not_ a browser-centric library.

Reply:
Versions should be passed into this service's constructors as to avoid passing in another service to query the needed values.

Reply:
I see, I was intending to set version to storage in web/mobile/desktop and get its value here like `deviceInterface.getRawStorageValue(key)`, so in case of mobile it would be `AsyncStorage`, for example. So the implementation would have been:
```
if (typeof localStorage !== 'undefined') {
  // browser or desktop - get the version from local storage
} else if (typeof AsyncStorage !== 'undefined') {
  // mobile - get the version from async storage
}
```

> Versions should be passed into this service's constructors
Makes sense, will update that way.